kafka/Dockerfile

```diff
@@ -1,19 +1,29 @@
 # https://kafka.apache.org/quickstart
 
-# https://catalog.redhat.com/software/containers/redhat-openjdk-18/openjdk18-openshift/58ada5701fbe981673cd6b10
-FROM registry.access.redhat.com/redhat-openjdk-18/openjdk18-openshift:1.12-1.1658775370
+# https://catalog.redhat.com/software/containers/ubi8/ubi/5c359854d70cc534b3a3784e
+FROM registry.access.redhat.com/ubi8/ubi:8.6 AS base
 
-ENV KAFKA_HOME "/opt/kafka"
+ENV KAFKA_VERSION=3.3.0
+ENV KAFKA_SITE_URL=https://dlcdn.apache.org/kafka/${KAFKA_VERSION}/kafka_2.13-${KAFKA_VERSION}.tgz
```

Comment:
I'm unable to run `make kafka-build` and it seems like it could be because this URL is invalid. If I curl this manually, I get a 404. Is it just me?

Reply:
no! it happened to me during coding this!! And I needed to figure out a way to avoid this happens, other way, this will happens everytime a new versioin is released, I completely forgot it.  :+1: :+1: 

Things that cross my mind:
- the value should passed as a build argument instead of a environment variable.
- Out of the Dockerfile, the right version still needs to be retrieved; I would like to avoid to extract the information by parsing the download page (https://kafka.apache.org/downloads), or (https://dlcdn.apache.org/kafka/).

So far, I will update the version from `3.3.0` to `3.3.1`; but I have to change this to make it dynamic.

Reply:
ah yeah, i did run into this and just updated it locally to 3.3.1.  I should have mentioned this :)  Shame that kafka doesn't include old versions.

Reply:
Another option instead of having every user build it themselves, is we host a built version in quay.

Reply:
update: I have added a hack by using a python script to retrieve the version, but it could fails when the html layout change; I retrieve the first h3 tag so far.